src/main/java/com/github/mc1arke/sonarqube/plugin/ce/pullrequest/bitbucket/BitbucketServerPullRequestDecorator.java

```diff
@@ -0,0 +1,235 @@
+        final AtomicInteger chunkCounter = new AtomicInteger(0);
+
+        client.deleteAnnotations(project, repo, analysisDetails.getCommitSha());
+
+        Map<Object, Set<Annotation>> annotationChunks = analysisDetails.getPostAnalysisIssueVisitor().getIssues().stream()
+                .filter(i -> i.getComponent().getReportAttributes().getScmPath().isPresent())
+                .filter(i -> i.getComponent().getType() == Component.Type.FILE)
+                .filter(i -> OPEN_ISSUE_STATUSES.contains(i.getIssue().status()))
+                .sorted(Comparator.comparing(a -> Severity.ALL.indexOf(a.getIssue().severity())))
+                .map(componentIssue -> {
+                    String path = componentIssue.getComponent().getReportAttributes().getScmPath().get();
+                    return new Annotation(componentIssue.getIssue().key(),
+                            Optional.ofNullable(componentIssue.getIssue().getLine()).orElse(0),
+                            analysisDetails.getIssueUrl(componentIssue.getIssue().key()),
+                            componentIssue.getIssue().getMessage(),
+                            path,
+                            toBitbucketSeverity(componentIssue.getIssue().severity()),
+                            toBitbucketType(componentIssue.getIssue().type()));
+                }).collect(Collectors.groupingBy(s -> chunkCounter.getAndIncrement() / DEFAULT_MAX_ANNOTATIONS, toSet()));
+
+        for (Set<Annotation> annotations : annotationChunks.values()) {
+            try {
+                client.createAnnotations(project, repo, analysisDetails.getCommitSha(), new CreateAnnotationsRequest(annotations));
+            } catch (BitbucketException e) {
+                if (e.isError(BitbucketException.PAYLOAD_TOO_LARGE)) {
+                    LOGGER.warn("The annotations will be truncated since the maximum number of annotations for this report has been reached.");
+                } else {
+                    throw e;
+                }
+
+            }
+        }
+    }
+
+    private String asInsightStatus(QualityGate.Status status) {
+        return QualityGate.Status.ERROR.equals(status) ? "FAIL" : "PASS";
+    }
+
+    private String toBitbucketSeverity(String severity) {
+        if (severity == null) {
+            return "LOW";
+        }
+        switch (severity) {
+            case Severity.BLOCKER:
+            case Severity.CRITICAL:
+                return "HIGH";
+            case Severity.MAJOR:
+                return "MEDIUM";
+            default:
+                return "LOW";
+        }
+    }
+
+    private String toBitbucketType(RuleType sonarqubeType) {
+        switch (sonarqubeType) {
+            case SECURITY_HOTSPOT:
+            case VULNERABILITY:
+                return "VULNERABILITY";
+            case CODE_SMELL:
+                return "CODE_SMELL";
+            case BUG:
+                return "BUG";
+            default:
+                return "";
```

Comment:
I'd throw in `IllegalStateException` rather than return an empty string given this would be an unexpected state.